### ISSUE TITLE
prediction format

### ISSUE THREAD

**Issue by mjanddy** [wondervictor/WiderFace-Evaluation#3]
what's the Format of prediction results？

**Comment by chi0tzp** [comment]
Could you provide an example of what does the <prediction directory> contain? Is this as described in the [official site](http://mmlab.ie.cuhk.edu.hk/projects/WIDERFace/)?

More specifically, they say that:

> The detection result for each image should be a text file, with the same name of the image. The detection results are organized by the event categories. For example, if the directory of a testing image is "./0--Parade/0_Parade_marchingband_1_5.jpg", the detection result should be writtern in the text file in "./0--Parade/0_Parade_marchingband_1_5.txt". The detection output is expected in the follwing format:
> ...
> < image name i >
> < number of faces in this image = im >
> < face i1 >
> < face i2 >
> ...
> < face im >
> ...
> Each text file should contain 1 row per detected bounding box, in the format "[left, top, width, height, score]". Please see the output example files and the README if the above descriptions are unclear. 
> 

So, if we want to evaluate a trained model on the validation set, should we just create one file for each image of the validation set (in the aforementioned format) and put all the files under a directory (which will be your <predictions dir>)?

Thanks for your time!

**Comment by KevinZhang1201** [comment]
you can see it at homepage of WIDER FACE and download Examples and Format of submissions.
Link: http://mmlab.ie.cuhk.edu.hk/projects/WIDERFace/support/example/Submission_example.zip
 
first you need to create a directory for all prediction results, like:
 
-- WFPred
       -- 0--Parade
               -- 0_Parade_marchingband_1_20.txt
               -- 0_Parade_marchingband_1_74.txt
               ...
       -- 1--Handshaking
       ...

In each files the detection results should be written as:
image_name
the number fo faces
x, y, w, h, confidence
...

**Comment by monkeyMirandar** [comment]
**

1. My well-trained model is a.pth file, and my well-tested model is a.pkl file. How to make the prediction format if you want to use this method for assessment?

**

**Comment by KevinZhang1201** [comment]
@monkeyMirandar Sorry, i'm not very sure what you mean. But I think you need to writte your own script to make the prediction on WIDER Face dataset with your own models and generate the prediction files as above mentioned.

**Comment by monkeyMirandar** [comment]
就是我不知道那个 prediction dir 里面应该放些什么


发自我的iPhone

------------------ Original ------------------

**Comment by KevinZhang1201** [comment]
Prediction dir
└─── 0--Parade
├───└─── 0_Parade_marchingband_1_20.txt
├───├─── 0_Parade_marchingband_1_74.txt
├───├─── ...
├─── 1--Handshaking
├───└─── 1_Handshaking_Handshaking_1_35.txt
├───├─── 1_Handshaking_Handshaking_1_94.txt
├───├─── ...
├─── ...
the files should be organized as this. 

![image](https://user-images.githubusercontent.com/44976087/62199850-8f22fa80-b384-11e9-90c2-55711afb4aa5.png)

![image](https://user-images.githubusercontent.com/44976087/62199938-b4b00400-b384-11e9-8e32-f96d29c07609.png)

**Comment by monkeyMirandar** [comment]
我的训练模型已经出来了！怎么才可以得到这个文件夹呀？可以具体说说吗？


发自我的iPhone

------------------ Original ------------------

**Comment by KevinZhang1201** [comment]
if you use python to test your model, you can write it like this:
![image](https://user-images.githubusercontent.com/44976087/62210396-5db62900-b39c-11e9-8981-440dc421a09f.png)

1. iterate all images in wider face dataset
2. apply your to each images, generate the prediction results and write them into cooresponding file

**Comment by foocker** [comment]
> 
> 我的训练模型已经出来了！怎么才可以得到这个文件夹呀？可以具体说说吗？ 发自我的iPhone
> […](#)
> ------------------ Original ------------------ From: KevinZhang1201 <notifications@github.com> Date: Wed,Jul 31,2019 5:17 PM To: wondervictor/WiderFace-Evaluation <WiderFace-Evaluation@noreply.github.com> Cc: monkeyMirandar <1391069856@qq.com>, Mention <mention@noreply.github.com> Subject: Re: [wondervictor/WiderFace-Evaluation] prediction format (#3) Prediction dir └─── 0--Parade ├───└─── 0_Parade_marchingband_1_20.txt ├───├─── 0_Parade_marchingband_1_74.txt ├───├─── ... ├─── 1--Handshaking ├───└─── 1_Handshaking_Handshaking_1_35.txt ├───├─── 1_Handshaking_Handshaking_1_94.txt ├───├─── ... ├─── ... the files should be organized as this. — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or mute the thread.

....

**Comment by foocker** [comment]
> 
> 
> if you use python to test your model, you can write it like this:
> ![image](https://user-images.githubusercontent.com/44976087/62210396-5db62900-b39c-11e9-8981-440dc421a09f.png)
> 
>     1. iterate all images in wider face dataset
> 
>     2. apply your to each images, generate the prediction results and write them into cooresponding file

the *.mat file are corresponding test dataset? why not val dataset. so,  evaluate on val dataset, you should rewrite evaluation code?

i have test on val dataset,  it's work!  but i'm not plan to read the evaluate.py .  so, what's logic of evaluate.py? such , 
  easy, midum, hard, in *.mat,  but val dataset has no these class, how it evaluate results  that preddicted from val dataset,  for *.mat easy,midum, hard class, but in *.mat.

**Comment by Hiteshsaai** [comment]
can the prediction results be in text format?, meanwhile the ground truth is in .mat format

**Comment by wondervictor** [comment]
prediction results are the same as the submission examples (txt files)

**Comment by huyutao3550346** [comment]
hello what is the format when we have multi-faces? 
could you give a .txt with multi-faces?

**Comment by NguyenKhacTuanAnh** [comment]
> prediction results are the same as the submission examples (txt files)

why do it have this error? I have run python setup.py build_ext --inplace
and python3 evaluation.py -p results/ -g ground_truth/

Reading Predictions : 100%|█████████████████████████████████████████████████████| 61/61 [00:00<00:00, 603.65it/s]
Processing easy: 100%|█████████████████████████████████████████████████████████| 61/61 [00:00<00:00, 4502.23it/s]
Processing medium: 100%|███████████████████████████████████████████████████████| 61/61 [00:00<00:00, 4714.26it/s]
Processing hard: 100%|█████████████████████████████████████████████████████████| 61/61 [00:00<00:00, 4092.99it/s]
==================== Results ====================
Easy   Val AP: 0.0 
Medium Val AP: 0.0 
Hard   Val AP: 0.0
